### PR TITLE
feature: optional retry to take a picture on error

### PR DESCRIPTION
**Pull request by andi34**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [X] New feature

#### What changes did you make? (Give an overview)
Add an option to retry taking a picture on error. This might be useful e.g. if gphoto fails because the fokus was not ready.

#### Is there anything you'd like reviewers to focus on?
~~As a next step, the api.cheese function should be revisited to avoid double code.~~

### CODE REVIEW

File: src/js/core.js [hunk 1]
```diff
@@ -421,7 +422,41 @@ const photoBooth = (function () {
...
+                            )
+                        );
+                        api.startCountdown(config.picture.retry_timeout, counter, () => {
+                            loading.empty();
+                            counter.empty();
+                            cheese.empty();
+
+                            if (config.picture.no_cheese) {
+                                photoboothTools.console.log('Cheese is disabled.');
+                                api.callTakePicApi(data, retry);
```
Comment (andi34) [inline src/js/core.js:443]: Pass `retry` to countdown, handle if statement to call `api.callTakePicApi` from there.

**Review by up-87 — APPROVED** [review]
Looks good to me. Used my lunch break to quickly test it on a photobooth and it works like expected (after I noticed and fixed the fact that I uninstalled gphoto2 in my recent experiments with python-gphoto - but that's my fault :D )

**Comment by andi34** [comment]
Thanks @up-87 , I am going to make mentioned changes to the api.cheesein the evening to avoid double code

File: lib/configsetup.inc.php [hunk 2]
```diff
@@ -227,6 +227,28 @@
             'name' => 'picture[preview_before_processing]',
             'value' => $config['picture']['preview_before_processing'],
         ],
+        'picture_retry_on_error' => [
```
Comment (andi34) [inline lib/configsetup.inc.php:230]: Keep in general section of better move to picture section?

**Comment by andi34** [comment]
Adminpanel: Keep in general section? Move to picture section?

File: lib/configsetup.inc.php (hunk 2 above)
Comment (andi34) [reply lib/configsetup.inc.php:230]: no, else we would have to add it to collage section too and also handle it there different

**Pull request merged**